src/Raven.Server/RavenServerStartup.cs

```diff
@@ -44,15 +45,50 @@ public void Configure(IApplicationBuilder app, ILoggerFactory loggerfactory)
 
             _router = app.ApplicationServices.GetService<RequestRouter>();
             _server = app.ApplicationServices.GetService<RavenServer>();
+            if (IsAccessUnauthorized())
+            {
+                app.Run(UnAuthorizedRequestHandler);
+            }
             app.Run(RequestHandler);
         }
 
+        private bool IsAccessUnauthorized()
+        {
+            if (_server.Configuration.Server.AnonymousUserAccessMode == AnonymousUserAccessModeValues.None)
+                return false;
+            if (_server.Configuration.Server.AllowAdminAnonymousAccessWhenNotbindedToLocalhost == true)
+                return false;
+            var url = _server.Configuration.Core.ServerUrl.ToLowerInvariant();
+            var uri = new Uri(url);
+            //url isn't set to localhost 
+            return !(uri.IsLoopback  || uri.Host == "0.0.0.0");
+        }
+
         public static bool SkipHttpLogging;
 
+        private Task UnAuthorizedRequestHandler(HttpContext context)
+        {
+            context.Response.StatusCode = (int) HttpStatusCode.ServiceUnavailable;
+            context.Response.Headers["Content-Type"] = "application/json; charset=utf-8";
+            JsonOperationContext ctx;
+            using (_server.ServerStore.ContextPool.AllocateOperationContext(out ctx))
+            using (var writer = new BlittableJsonTextWriter(ctx, context.Response.Body))
+            {
+                writer.WriteStartObject();
+                writer.WritePropertyName("Message");
+                writer.WriteString($"The server is running in an un-authorized mode.{Environment.NewLine}" +
```

Comment:
This goes as JSON string, so it will not be as readable with the line breaks escaped
Make this into an array as well, so when formatted, it will be easier to read. (The array is just for display purposes, nothing more.